File: 2016/7. Internet Protocol Version 7/1.Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int ans = 0;
    ifstream input("input.txt");
    string ip;
    while (getline(input, ip)) {
        bool fcond = false, scond = true;
        int j = 0;
        while (j < ip.size() - 3) {
            for (; j < ip.size() - 3; j++) {
                if (ip[j] == ip[j+3] and ip[j+1] == ip[j+2] and ip[j] != ip[j+1]) {
                    fcond = true;
                } else if (ip[j] == '[') {
                    j++;
                    break;
                }
            }
            for (; j < ip.size() - 3; j++) {
                if (ip[j] == ip[j+3] and ip[j+1] == ip[j+2] and ip[j] != ip[j+1]) {
                    scond = false;
                } else if (ip[j] == ']') {
                    j++;
                    break;
                }
            }
        }
        if (fcond && scond) ans++;
    }
    
    cout << ans;
}

File: 2017/3. Spiral Memory/1.Solution.py
n = int(input())

s = 1
while n > s * s:
    s += 2

if n == s * s:
    ans = s - 1
    print(ans)
    exit()

ans = s // 2

candid = s * s - s // 2
while abs(n - candid) > s // 2:
    candid -= s - 1

ans += abs(n - candid)
print(ans)
